### PR TITLE
NameTags rewrite

### LINKED ISSUE #673: Rewrite nametags in support of additional features and bug fixes
See #12 or zeroeightysix/KAMI/issues/4, I've been wanting to do this for a while but never wrote it down :p 

I'd do this purely for more experience with OpenGL I think, but don't expect it for a while 

- [ ] proper scaling for distance from players (like Future)
- [ ] fix enchantments glint breaking sometimes
- [ ] item name and durability
- [ ] player ping
- [ ] existing features such as showing enchantments and held items 
  - [ ] option to store x last held items

### LINKED ISSUE #1157: Nametags armor doesn't display through walls

### PR DESCRIPTION
**Pull request by Xiaro**
**Describe the pull**
1. Rewrote NameTags entirely with fully (at least mostly) customizable settings
2. Closes #1157 
3. Doesn't closes #673 completely because it doesn't store last held items (I don't think that is n

**Additional context**
See Discord for screenshots

### CODE REVIEW

File: src/main/java/me/zeroeightsix/kami/module/modules/render/Nametags.kt [hunk 1]
```diff
@@ -0,0 +1,529 @@
...
+        } else {
+            if (filled.value)
+                RenderUtils2D.drawRoundedRectFilled(vertexHelper, posBegin, posEnd, cornerRadius.value.toDouble(), 8, ColorHolder(rFilled.value, gFilled.value, bFilled.value, aFilled.value))
+            if (outline.value && outlineWidth.value != 0f)
+                RenderUtils2D.drawRoundedRectOutline(vertexHelper, posBegin, posEnd, cornerRadius.value.toDouble(), 8, outlineWidth.value, ColorHolder(rOutline.value, gOutline.value, bOutline.value, aOutline.value))
+        }
+    }
+
+    override fun onUpdate() {
+        if (timer.tick(5L) && checkSetting()) MessageSendHelper.sendChatMessage("$chatName Totem pops is not yet implemented")
```
Comment (dominikaaaa) [inline src/main/java/me/zeroeightsix/kami/module/modules/render/Nametags.kt:284]: don't add the setting if not added yet?

**Review by dominikaaaa — CHANGES_REQUESTED** [review]
Looks good, just remove the totem pop setting and I'll merge. Readd the setting once it's ready

File: src/main/java/me/zeroeightsix/kami/module/modules/render/Nametags.kt (hunk 1 above)
Comment (Xiaro) [reply src/main/java/me/zeroeightsix/kami/module/modules/render/Nametags.kt:284]: Well just leave as as a TODO thing there lol. So someone sees it and they can impl it

Comment (dominikaaaa) [reply src/main/java/me/zeroeightsix/kami/module/modules/render/Nametags.kt:284]: a comment in the code is fine. dead settings aren't fine

**Review by dominikaaaa — APPROVED** [review]

**Pull request merged**